sequencing/ATACSeq/jobSubmission/0_setUp.sh

```diff
@@ -0,0 +1,174 @@
+mkdir -p $LOG_DIR
+mv ATACSetUpS0-${SLURM_JOB_ID}* $LOG_DIR
+
+## ============ ##
+##    CHECKS    ##
+## ============ ##
+
+## CONFIG FILE ##
+
+## Check if config file for input project exist
+
+if [[ -s "/lustre/projects/Research_Project-MRC190311/ATACSeq/$1/config.txt" ]]
+then
+  source "/lustre/projects/Research_Project-MRC190311/ATACSeq/$1/config.txt"
+  echo "Config file for $1 project found."
+else 
+  { echo "Config file for chosen project does not exist or is in the wrong directory. Please create this first." ; exit 1; }  
+fi
+
+## Check if R config file for input project exist
+
+if [[ -s "/lustre/projects/Research_Project-MRC190311/ATACSeq/$1/config.r" ]]
+then
+  echo "R config file for $1 project found."
+else 
+  { echo "R config file for chosen project does not exist or is in the wrong directory. Please create this first." ; exit 1; }  
+fi
+
+## DIRECTORIES ##
+
+## Check that the metadata directory exists and contains sample list.
+if [[ ! -d ${METADIR} ]]
+then
+	echo 'METADATA directory does not exist'
+	exit 1
+elif  test -f ${METADIR}/samples.txt; then
+	mapfile -t SAMPLEIDS < ${METADIR}/samples.txt 
+	echo "Sample list found. Number of sample IDs found:" """${#SAMPLEIDS[@]}"""
+else
+	echo 'samples.txt file does not exist in METADIR'
+fi
+
+## Check that the sampleSheet exist
+if [[ -d ${METADIR} ]] && [[ -s "${METADIR}/sampleSheet.csv" ]]
+then
+  SAMPLEIDS2=( $(tail -n +2 ${METADIR}/sampleSheet.csv  | cut -d ',' -f2) )
+  echo "Sample sheet found. Number of sample IDs found:"" ""${#SAMPLEIDS2[@]}"""
+else
+  echo "sampleSheet.csv file does not exist in METADIR."
+fi
+
+## Check if remaining main directories exist and create if not
+dirs=(${RAWDATADIR} ${FASTQCDIR} ${TRIMDIR} ${ALIGNEDDIR} ${PEAKDIR})
+type=("DATA" "FASTQC" "TRIMMED" "ALIGNED" "PEAK CALLED")
+
+for (( x=0; x<${#dirs[@]}; x++ ))
+do
+	if [ -d ${dirs[x]} ] 
+	then 
+		echo ${type[x]} "directory:" ${dirs[x]}
+	elif [ ${dirs[x]} = $RAWDATADIR ]; then 
+		echo ${dirs[x]} "directory does not exist"
+		exit 1
+	else
+		echo "Creating directory: " ${dirs[x]}
+		mkdir -p ${dirs[x]} 
+	fi
+done
+
+## MODULES ##
+
+## Check that the SAMstats environment exists and create it if not
+
+module load Anaconda3
+if [ -d ${ENVDIR} ]
+then 
+	echo "ENCODEQC environment exists"
+  source ${CONDAENV}
+  conda activate ${ENVDIR}
+  echo "List of packages included in conda environment is output in project directory"
+  conda list > "/lustre/projects/Research_Project-MRC190311/ATACSeq/$1/condaEnv.txt"
+  conda deactivate
+else
+	echo "ENCODEQC environment does not exist"
+	echo "Creating environment"
+  source ${CONDAENV}
+	conda create --prefix ${ENVDIR}
+	conda activate ${ENVDIR}
+	echo "Installing SAMstats"
+	conda install -c bioconda samstats
+	conda deactivate
+	echo "ENCODEQC environment has been created and SAMstats installed"
+fi
+
+## Check that pip environment exist and create it if not
+module load Python/3.9.6-GCCcore-11.2.0-bare
```

Comment:
Can we collate all the module load statements at the top? Again this assumes the user is on a HPC system with a python module with this name.

Reply:
Could we do all of this through condo and minimise the number of module loads?